extensions/assistant/deployment-spi/src/main/java/io/quarkus/assistant/deployment/spi/Assistant.java

```diff
@@ -0,0 +1,54 @@
+package io.quarkus.assistant.deployment.spi;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+
+/**
```

Comment:
Regarding API, @geoand and @mariofusco use "records" to encapsulate the parameters. At some point, we will likely add more parameters (typically, output schema, maybe guardrails...). I wonder if we should do the same to avoid having methods with many parameters and overloading.

Reply:
+1.

This is done because like @cescoffier alluded to, it's very hard to know at the SPI creation time what exact options will be neede, so by introducing an "object" to encapsulate all params, we can add new ones in the future without breaking existing SPI implementations.
Furthermore, for maximum flexibility, using a interface instead of a record is best for these parameters, but as they are meant to be only be consumed (and are created by the caller of the SPI, hence under our complete control), it should't make a difference.

Reply:
@cescoffier let's discuss this, I think the way I have done it allows for the same